### DISCUSSION TITLE
Can't push multiple policies to S3 policy

### DISCUSSION THREAD

**Discussion by rajandmr** [aws/aws-cdk#22563] [Q&A]
Hey Guys, I've been trying to add multiple OAI policies inside the S3. We have a core stack for creating CloudFront and other configs and we create 3 other stacks using the Core stack. I'm adding the following code but at the end, I'm always ending up getting just one policy inside the S3 Policy.

```
const cloudfrontOAI = new cloudfront.OriginAccessIdentity(
      this,
      `${name}-cloudfront-OAI`,
      {
        comment: `OAI for ${name}`,
      },
    );

    // S3 Buckets
    const myBucket = s3.Bucket.fromBucketName(
      this,
      'MyBucket',
      props.bucketName,
    );

const policyStatement = new iam.PolicyStatement({
      actions: ['s3:GetObject'],
      resources: [
        siteBucket.arnForObjects('*'),
      ],
      principals: [cloudfrontOAI.grantPrincipal],
    });

    const bucketPolicy = new s3.BucketPolicy(
      this,
      `${name}-cloudfrontAccessBucketPolicy`,
      {
        bucket: myBucket,
      },
    );
    bucketPolicy.document.addStatements(policyStatement);

```
Any help would be greatly appreciated, been stuck in this issue for a week.

**Comment by pahud** [comment]
You can run `cdk synth` and check out the synthesized YAML. It appears to me, according to your code, you are just adding one policyStatement in a new BucketPolicy for an existing bucket. I didn't see two policyStatements from your code above.
